src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -4179,6 +4179,28 @@ object Desugar extends LazyLogging {
             } yield in.PredExprUnfold(predExpInstance.base.asInstanceOf[in.PredicateConstructor], args, access.p)(src)
             case _ => for {e <- goA(exp)} yield in.Unfold(e.asInstanceOf[in.Access])(src)
           }
+        case PDeriveStmt(exp, block) =>
+          /**
+            * `derive cond by block`
+            * is encoded to
+            * ```
+            * if !cond {
+            *   block
+            *   assert false
+            * }
+            * ```
+            */
+          for {
+            cond <- exprD(ctx, info)(exp)
+            b <- stmtD(ctx, info)(block)
+            // transform error of `assert false` to indicate that the derive statement failed
+            annotatedSrc = Source.Parser.Single(stmt, Source.AnnotatedOrigin(src.origin.get, new Source.OverwriteErrorAnnotation(_ match {
+              case e: AssertError => DeriveError(e.info)
+              case e => e
+            }, false)))
```

Comment:
This line is too long. I would probably move part of this (e.g., the case distinction on the error) to another line to make it easier to read.